### ISSUE TITLE
Search handles initials incorrectly

### ISSUE THREAD

**Issue by tfmorris** [internetarchive/openlibrary#7035]
There are a variety of problems with the way that search handles initials. The three most annoying are:

1. Failure to tokenize initials which aren't separated by spaces correctly.
2. Failure to consider order of initials when ranking search results
3. Inability to search for repeated initials

The results for these two queries should be the same:
https://openlibrary.org/search/authors?q=h.a.j. (9 results)
https://openlibrary.org/search/authors?q=h.+a.+j. (254 results)

This query returns one of the worst matches first and pushes exact matches well down the list:
https://openlibrary.org/search/authors?q=e.+h.+smith

This query has an exact match but returns 50 extraneous results before getting to it (OL1661733A):
https://openlibrary.org/search/authors?q=h.+h.+daniel 

### Steps to Reproduce

1. Go to ...
5. Do ...

* Actual:
* Expected:

### Details

- **Logged in (Y/N)?**
- **Browser type/version?**
- **Operating system?**
- **Environment (prod/dev/local)?** prod

**Comment by tfmorris** [comment]
@mekarpeles @cdrini Do you need more details or examples before this can be triaged? Please let me know and I'll see what I can provide.

**Comment by cdrini** [comment]
No, I think I got it! Hmm hairy one. The best thing might be to add phrase slop to solr, so a search for `h a j` will boost matches for `"h a j"` above matches for `h OR a OR j`.

**Comment by cdrini** [comment]
Likely here will want to add it:

https://github.com/internetarchive/openlibrary/blob/14f8029ff74eb34c25c5cf7262f5662d655347fc/openlibrary/plugins/worksearch/schemes/authors.py#L44-L55

See https://solr.apache.org/guide/8_11/the-extended-dismax-query-parser.html#using-slop for parameters to add.

**Comment by cdrini** [comment]
This should be done deploying to testing.openlibrary.org in a few mins. Please let me know if that made things better!

**Comment by cdrini** [comment]
Seems to have had no impact... 🤔

**Comment by cdrini** [comment]
Ah because by default we sort number of works, which make this moot.

**Comment by cdrini** [comment]
Ok changed sorting to be by relevance, and only boost work_count instead of sorting by it. It's up on testing. Let me know if that does the trick!

**Comment by cdrini** [comment]
Ah interesting it looks like solr's standard tokenizer doesn't split on `.`:

> Periods (dots) that are not followed by whitespace are kept as part of the token, including Internet domain names.

https://solr.apache.org/guide/8_11/tokenizers.html

**Comment by cdrini** [comment]
And you're exactly correct; under the hood `h. h. daniel` gets converted to a query like `name:(h OR h OR dnaiel) name:"h h daniel"^10` effecitvely.

**Comment by cdrini** [comment]
Separated off the chunk for better ranking into its own issue. We're aiming for "Milestone 0"'s this year -- ie at the end of every month, everything on the milestone should be complete -- as an exercise to improve project management. I won't be able to tackle the tokenization issues this month, so pulled out just the re-ranking into its own issue.
